libminifi/include/core/yaml/YamlConfiguration.h

```diff
@@ -88,10 +102,17 @@ class YamlConfiguration : public FlowConfiguration {
   void parseRemoteProcessGroupYaml(YAML::Node *node,
                                    core::ProcessGroup * parent);
   // Process Provenance Report YAML
-  void parseProvenanceReportingYaml(YAML::Node *reportNode, core::ProcessGroup * parentGroup);
+  void parseProvenanceReportingYaml(YAML::Node *reportNode,
+                                    core::ProcessGroup * parentGroup);
   // Parse Properties Node YAML for a processor
   void parsePropertiesNodeYaml(YAML::Node *propertiesNode,
                                std::shared_ptr<core::Processor> processor);
+  // Helper function to get or generate a component's id field
```

Comment:
I like that. I'll update all existing function comments in YamlConfiguration.h while I'm here